### PR TITLE
Fix test server for chunked data

### PR DESCRIPTION
**Pull request by imurchie**
On Travis the data often gets chunked and causes flakiness. Simply cache the chunk and wait for the next one.

### CODE REVIEW

File: test/helpers/remote-debugger-server.js [hunk 1]
```diff
@@ -272,14 +272,28 @@ class RemoteDebuggerServer {
...
         c.on('data', (data) => {
-          let plist = bplistParse.parseBuffer(data.slice(4));
+          if (leftOverData) {
+            data = Buffer.concat([leftOverData, data]);
+            leftOverData = null;
+          }
+
+          let plist;
+          try {
+            plist = bplistParser.parseBuffer(data.slice(4));
```
[the remark this answers is not in the document]
Comment (imurchie) [reply test/helpers/remote-debugger-server.js]: It's magic, I guess. It's where the binary plist begins.

File: test/helpers/remote-debugger-server.js [hunk 2]
```diff
@@ -272,14 +272,28 @@ class RemoteDebuggerServer {
...
+          } catch (err) {
+            log.info(`Unable to decipher plist: ${err.message}`);
+            log.info('Assuming just a chunk, waiting for more');
```
[the remark this answers is not in the document]
Comment (imurchie) [reply test/helpers/remote-debugger-server.js]: No need. There is nothing that can be done if it doesn't get called. And if it does wait forever the test will fail with the mocha timeout, which is as much as we could do anyway.

**Comment by imurchie** [comment]
Ran the tests 12 times with no errors.

File: test/helpers/remote-debugger-server.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply test/helpers/remote-debugger-server.js]: perhaps it would be more readable to put it into a constant

**Pull request merged**